app/models/topic_tracking_state.rb

```diff
@@ -549,12 +549,13 @@ def self.trigger_post_read_count_update(post, groups, last_read_post_number, use
   end
 
   def self.secure_category_group_ids(topic)
-    ids = topic&.category&.secure_group_ids
+    category = topic.category
 
-    if ids.blank?
-      [Group::AUTO_GROUPS[:admin]]
+    if category.read_restricted
+      ids = [Group::AUTO_GROUPS[:admins]]
+      ids.push(*category&.secure_group_ids)
```

Comment:
The save nav should be up on line 554 here

Reply:
Sorry I just pushed a change to remove the safe navigation operator since a regular topic will always have a category.

Reply:
Oh true, I thought there was a possibility it could be nil. All good :+1: